src/logic/embedding.lean

```diff
@@ -93,6 +103,14 @@ protected noncomputable def equiv_of_surjective {α β} (f : α ↪ β) (hf : su
   α ≃ β :=
 equiv.of_bijective f ⟨f.injective, hf⟩
 
+/-- An equivalence between all injective functions and all embeddings. -/
+def equiv_inj_subtype (α : Sort u) (β : Sort v)
+  : {f : α → β // function.injective f} ≃ (α ↪ β) :=
+{ to_fun := λ f, ⟨f.val, f.property⟩,
+  inv_fun := λ f, ⟨f, f.injective⟩,
+  left_inv := λ f, by simp,
+  right_inv := λ f, by {ext, simp} }
```

Comment:
This feels like a weird definition to have; mathlib has a lot of types that are just a single data field with some proof fields, but we don't normally bother to construct equivalences with subtypes. Perhaps it's worth just constructing this locally in the file it's needed, and making it `private`?

Reply:
dope, wasn't sure whether this would be of general use